lib/matplotlib/backends/backend_qtagg.py

```diff
@@ -47,25 +46,20 @@ def paintEvent(self, event):
             right = left + width
             # create a buffer using the image bounding box
             bbox = Bbox([[left, bottom], [right, top]])
-            buf = memoryview(self.copy_from_bbox(bbox))
```

Comment:
I think it would work to simply do `buf = memoryview(np.asarray(self.copy_from_bbox(bbox)))` here and not touch anything else?  This way from the pure python side you just need something that implements `__array__` (a standard numpy interface) and there's no need to introduce a new custom API (get_bounds).

See also https://github.com/matplotlib/matplotlib/issues/23882 and the linked discourse thread.

Reply:
Thanks for the linked issue, very relevant. FWIW, I'm fully in favour of removing ``BufferRegion`` (a non-public API), but not so hot on removing ``restore_region`` (a public API). If ``restore_region`` is simply redirected to ``draw_image`` then fair enough, though I don't especially like the idea of ``gc=None``, nor changing the blitting blending model.

That linked discussion was super interesting - I had assumed that there was nobody using a subset bbox for ``restore_region`` - it seems like an obscure API to me. I was going to propose it be removed - I would still do that personally (and make it easy to subset the copied region instead). 

Ultimately, I think we share the same objective here... I'm guessing this is about standardising the (rasterising) renderer such that in the future we could have backends which are not renderer specific (e.g. a common implementation of a qt backend for both agg and cairo (and other potential renderers)).

Reply:
I took a look at this, and you are right. I like it because it avoids extending the existing inaccessible type. It doesn't solve the fact that we do need to return more than just an image (we need the origin of the image too), but I'm confident we could remove ``BufferRegion`` at this point and replace it with a simple Python object (currently a numpy subclass). Ideally, it would not need to be a numpy subclass in the future (containment over inheritance, and all)

Reply:
I should say: I don't see a need for ``memoryview`` at all. I just used your ``asarray`` trick.